lib/utils/note-utils.js

```diff
@@ -42,34 +62,13 @@ export const defaults = {
  */
 export const noteTitleAndPreview = note => {
   const content = (note && note.data && note.data.content) || '';
-
-  const match = isMarkdown(note)
-    ? noteTitleAndPreviewMdRegExp.exec(content || '')
-    : noteTitleAndPreviewRegExp.exec(content || '');
-
-  if (!match) {
-    return defaults;
-  }
-
-  let [, title = defaults.title, preview = defaults.preview] = match;
-
-  title = title.slice(0, titleCharacterLimit);
-  preview = preview.slice(0, previewCharacterLimit);
-
-  if (preview && isMarkdown(note)) {
-    // Workaround for a bug in `remove-markdown`
-    // See https://github.com/stiang/remove-markdown/issues/35
-    const previewWithSpacesTrimmed = preview.replace(/(\s)\s+/g, '$1');
-
-    preview = removeMarkdown(previewWithSpacesTrimmed, {
-      stripListLeaders: false,
-    });
-  }
-
+  const isNoteMarkdown = isMarkdown(note);
+  const title = getTitle( content, isNoteMarkdown );
+  const preview = getPreview( content, isNoteMarkdown );
```

Comment:
in the previous code we had somewhat more interleaved the computation of the title and preview but now they are separated. I wonder if instead of passing in this boolean flag and crossing layers of abstraction it would make more sense to let `getTitle` and `getPreview` focus specifically on the text, since it looks like all we're doing is transforming the output, not changing how we find it…

```js
const transformOutput = isMarkdown
  ? ( s => removeMarkdownWithFix( s ) || s )
  : s => s;

const title = transformOutput( getTitle( content ) );
const preview = transformOutput( getPreview( content ) );
```

the `getTitle` and `getPreview` functions would then collapse into a few lines each and we wouldn't have `rawTitle` and `title` and that branching code.

not sure what a good name for `transformOutput` would be, though with `lodash` we could approach it differently yet again

```js
const title = getTitle( content );
const preview = getPreview( content );

return isMarkdown
  ? mapValues( { title, preview }, s => removeMarkdownWithFix( s ) || s )
  : { title, preview };
```